go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -469,3 +428,45 @@ func (c *addressManager) sync() {
 		}
 	}
 }
+
+// updateOVNEncapIP updates encap IP to OVS when the node primary IP changed.
+func updateOVNEncapIPAndReconnect(newIP net.IP) {
+	checkCmd := []string{
+		"get",
+		"Open_vSwitch",
+		".",
+		"external_ids:ovn-encap-ip",
+	}
+	encapIP, stderr, err := util.RunOVSVsctl(checkCmd...)
+	if err != nil {
+		klog.Warningf("Unable to retrieve configured ovn-encap-ip from OVS: %v, %q", err, stderr)
+	} else {
+		encapIP = strings.TrimSuffix(encapIP, "\n")
+		if len(encapIP) > 0 && newIP.String() == encapIP {
+			klog.V(4).Infof("Will not update encap IP, value: %s is the already configured", newIP)
+			return
+		}
+	}
+
+	confCmd := []string{
+		"set",
+		"Open_vSwitch",
+		".",
+		fmt.Sprintf("external_ids:ovn-encap-ip=%s", newIP),
+	}
+
+	_, stderr, err = util.RunOVSVsctl(confCmd...)
+	if err != nil {
+		klog.Errorf("Error setting OVS encap IP: %v  %q", err, stderr)
```

Comment:
while your here can you also add the new IP to the err msg - might be helpful for debugging:
```
klog.Errorf("Error setting OVS encap IP %s: %v  %q", newIP.String(), err, stderr)
```